jquery.flot.composeImages.js

```diff
@@ -0,0 +1,171 @@
+(function($) {
+    "use strict";
+
+    function copyCanvasToImg(canvas, img) {
+        img.src = canvas.toDataURL('image/png');
+    }
+
+    function getCSSRules(document) {
+        var styleSheets = document.styleSheets,
+            rulesList = [];
+        for (var i = 0; i < styleSheets.length; i++) {
+            // in Chrome the external CSS files are empty when the page is loaded from directly disk
+            var rules = styleSheets[i].cssRules || [];
+            for (var j = 0; j < rules.length; j++) {
+                var rule = rules[j];
+                rulesList.push(rule.cssText);
+            }
+        }
+        return rulesList;
+    }
+
+    function embedCSSRulesInSVG(rules, svg) {
+        var text = [
+            '<svg width="' + svg.width.baseVal.value + '" height="' + svg.height.baseVal.value + '" viewBox="0 0 ' + svg.width.baseVal.value + ' ' + svg.height.baseVal.value + '" xmlns="http://www.w3.org/2000/svg">',
+            '<style>',
+            '/* <![CDATA[ */',
+            rules.join('\n'),
+            '/* ]]> */',
+            '</style>',
+            svg.innerHTML,
+            '</svg>'
+        ].join('\n');
+        return text;
+    }
+
+    function copySVGToImgMostBrowsers(svg, img) {
+        var rules = getCSSRules(document),
+            text = embedCSSRulesInSVG(rules, svg),
+            blob = new Blob([text], {type: "image/svg+xml;charset=utf-8"}),
+            domURL = self.URL || self.webkitURL || self,
+            url = domURL.createObjectURL(blob);
+        img.src = url;
+    }
+
+    function copySVGToImgSafari(svg, img) {
+        var rules = getCSSRules(document),
+            text = embedCSSRulesInSVG(rules, svg),
+            data = "data:image/svg+xml;base64," + btoa(text);
+        img.src = data;
+    }
+
+    function copySVGToImg(svg, img) {
+        // *** https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser
+        // Safari 3.0+ "[object HTMLElementConstructor]"
+        var isSafari = /constructor/i.test(window.HTMLElement) || (function (p) { return p.toString() === "[object SafariRemoteNotification]"; })(!window['safari'] || (typeof safari !== 'undefined' && safari.pushNotification));
+
+        if (isSafari) {
+            copySVGToImgSafari(svg, img);
+        } else {
+            copySVGToImgMostBrowsers(svg, img);
+        }
+    }
+
+    function prepareImagesToBeComposed(sources, destination) {
+        var Result = 0;
```

Comment:
Please rename Result to result to keep a consistent way of naming variables

Reply:
old habits..